clippy_lints/src/rc_clone_in_vec_init.rs

```diff
@@ -47,27 +49,107 @@ declare_lint_pass!(RcCloneInVecInit => [RC_CLONE_IN_VEC_INIT]);
+        let Some(VecArgs::Repeat(elem, len)) = VecArgs::hir(cx, expr) else { return; };
         let Some(symbol) = new_reference_call(cx, elem) else { return; };
+        let lint_span = macro_call.span;
+        let symbol_name = symbol.as_str();
+        let len_snippet = snippet(cx, len.span, "..");
+        let elem_snippet = elem_snippet(cx, elem, symbol_name);
+        let indentation = indent_of(cx, lint_span).unwrap_or(0);
+        let lint_suggestions =
+            construct_lint_suggestions(lint_span, symbol_name, &elem_snippet, len_snippet.as_ref(), indentation);
 
-        emit_lint(cx, symbol, &macro_call);
+        emit_lint(cx, symbol, lint_span, &lint_suggestions);
     }
 }
 
-fn emit_lint(cx: &LateContext<'_>, symbol: Symbol, macro_call: &MacroCall) {
+struct LintSuggestion {
+    span: Span,
+    message: String,
+    suggestion: String,
+    applicability: Applicability,
+}
+
+impl LintSuggestion {
+    fn span_suggestion(&self, diag: &mut Diagnostic) {
+        diag.span_suggestion(self.span, &self.message, &self.suggestion, self.applicability);
+    }
+}
+
+fn construct_lint_suggestions(
+    span: Span,
+    symbol_name: &str,
+    elem_snippet: &str,
+    len_snippet: &str,
+    indentation: usize,
+) -> Vec<LintSuggestion> {
+    let indentation = " ".repeat(indentation);
+    let loop_init_suggestion = loop_init_suggestion(elem_snippet, len_snippet, &indentation);
+    let extract_suggestion = extract_suggestion(elem_snippet, len_snippet, &indentation);
+
+    vec![
+        LintSuggestion {
+            span,
+            message: format!("consider initializing each `{symbol_name}` element individually"),
+            suggestion: loop_init_suggestion,
+            applicability: Applicability::Unspecified,
+        },
+        LintSuggestion {
+            span,
+            message: format!(
+                "or if this is intentional, consider extracting the `{symbol_name}` initialization to a variable"
+            ),
+            suggestion: extract_suggestion,
+            applicability: Applicability::Unspecified,
+        },
+    ]
+}
+
+fn elem_snippet(cx: &LateContext<'_>, elem: &Expr<'_>, symbol_name: &str) -> String {
+    let mut elem_snippet = snippet(cx, elem.span, "..").to_string();
+    if elem_snippet.contains('\n') {
+        let reference_initialization = format!("{symbol_name}::new");
+        // This string must be found in `elem_snippet`, otherwise we won't be constructing the snippet in
+        // the first place.
+        let reference_initialization_end =
+            elem_snippet.find(&reference_initialization).unwrap() + reference_initialization.len();
+        elem_snippet.replace_range(reference_initialization_end.., "..");
+    }
+    elem_snippet
+}
+
+fn loop_init_suggestion(elem: &str, len: &str, indent: &str) -> String {
+    format!(
+        r#"{{
+{indent}{indent}let mut v = Vec::with_capacity({len});
```

Comment:
Having two indents like this, will probably look weird, if the indent is != 4. Did you mean to write 
```suggestion
{indent}    let mut v = Vec::with_capacity({len});
```

Reply:
mmm I think it'll be better to show the same indentation for the suggestion's code block as the rest of the file. Why not aligning my indentation with the user's?

Reply:
The indention is correct, in this case you want to use the base indention plus the four for the block. If you set `indent` to `            ` (12 Spaces) then the suggestion  would look like this:

```rs
LL ~            let v1 = {
LL +                        let data = Arc::new(..);
LL +                        vec![data; 2]
LL ~            };
```

Instead, we only want the first indention to adapt and the second one to be ident + 4

Reply:
good point, fixed it now 🙂